Sources/Internal/Transformations/InputParser.cpp

```diff
@@ -13,4 +13,202 @@ namespace DataAnalysis { namespace Transformations {
   	}
   }
 
+  shared_ptr<IFunction<double>> GetSubFunctionAndInitialize( 
+	  __in const FUNCTION_TYPE type, 
+	  __in const size_t paramCount, 
+	  __in_ecount( paramCount ) const double *pParams, 
+	  __in_z const char *pFunctName,
+	  __in const InputTransformation *pCaller = nullptr ) 
+  {
+	  shared_ptr<IFunction<double>> spFunct = nullptr;
+
+	  uint argCount = paramCount - 1;
+	  const double *pArgs = pParams + 1;
+
+	  switch ( type )
+	  { 
+	  case ( FT_POLY_BASIC ):
+	  {
+		  spFunct = shared_ptr<IFunction<double>>( new BasicPolynomialTransform<double>() );
+		  _ASSERT( argCount > 0 );
+		  if ( strncmp( "XScl", pFunctName, 4 ) == 0 ) {
+			  spFunct->Initialize<BasicPolynomialTransform<double>>( paramCount, pParams ); // don't even ask, I'm about to cry
+		  }
+		  else {
+			  spFunct->Initialize<BasicPolynomialTransform<double>>( argCount, pArgs );
+		  }
+		  break;
```

Comment:
tam ide o to, ze pri XScl, polynomialnej fcii, nemas typ, ale je to napevno basic polynom (because logic...)